src/segment-loader.js

```diff
@@ -47,6 +47,30 @@ const detectEndOfStream = function(playlist, mediaSource, segmentIndex) {
 
 const finite = (num) => typeof num === 'number' && isFinite(num);
 
+export const illegalMediaSwitch = (loaderType, startingMedia, newSegmentMedia) => {
+  // Although these checks should most likely cover non 'main' types, for now it narrows
+  // the scope of our checks.
+  if (loaderType !== 'main' || !startingMedia || !newSegmentMedia) {
+    return;
```

Comment:
Since you make use of the return value, it might be a good idea to explicitly return `null` or `false`